src/libraries/sendtohelix.proj

```diff
@@ -1,122 +1,172 @@
+  <PropertyGroup>
+    <!-- Set the name of the scenario file. Note that this is only used in invocations where $(Scenario) is set
+         (which is when this project is invoked to call the "CreateOneScenarioTestEnvFile" target).
     -->
-    <HelixCommand Condition="'$(TargetOS)' == 'Windows_NT'">call RunTests.cmd --runtime-path %HELIX_CORRELATION_PAYLOAD%</HelixCommand>
-    <HelixCommand Condition="'$(TargetOS)' != 'Windows_NT'">./RunTests.sh --runtime-path "$HELIX_CORRELATION_PAYLOAD"</HelixCommand>
+    <TestEnvFileName Condition=" '$(TargetsWindows)' == 'true' ">SetStressModes_$(Scenario).cmd</TestEnvFileName>
+    <TestEnvFileName Condition=" '$(TargetsWindows)' != 'true' ">SetStressModes_$(Scenario).sh</TestEnvFileName>
   </PropertyGroup>
 
+  <Target Name="CreateOneScenarioTestEnvFile">
+    <!-- This target creates one __TestEnv file for the single $(Scenario). -->
+
+    <Error Condition="'$(Scenario)' == ''" Text="No Scenario specified" />
+
+    <PropertyGroup>
+      <TestEnvFilePath>$(TestHostRootPath)$(TestEnvFileName)</TestEnvFilePath>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_ProjectsToBuild Include="$(RepoRoot)\src\coreclr\tests\testenvironment.proj">
+        <Properties>Scenario=$(Scenario);TestEnvFileName=$(TestEnvFilePath);TargetsWindows=$(TargetsWindows)</Properties>
+      </_ProjectsToBuild>
+    </ItemGroup>
+
+    <Message Importance="High" Text="Creating $(TestEnvFilePath) for scenario $(Scenario)" />
+
+    <MSBuild Projects="@(_ProjectsToBuild)" Targets="CreateTestEnvFile" StopOnFirstFailure="true" />
+
+    <Error Condition="!Exists('$(TestEnvFilePath)')" Text="File $(TestEnvFilePath) not found!" />
+  </Target>
+
+  <Target Condition="'$(Scenarios)' != ''" Name="CreateAllScenarioTestEnvFiles">
+    <!-- This target creates one __TestEnv file for each of the scenarios in the $(Scenarios) comma-separated list. -->
+
+    <Message Importance="High" Text="Creating per-scenario TestEnv files for scenarios $(Scenarios)" />
+
+    <ItemGroup>
+      <_Scenario Include="$(Scenarios.Split(','))" />
+      <_ProjectsToBuild Include="$(MSBuildProjectFile)">
+        <AdditionalProperties>Scenario=%(_Scenario.Identity)</AdditionalProperties>
+      </_ProjectsToBuild>
+    </ItemGroup>
+
+    <MSBuild Projects="@(_ProjectsToBuild)" Targets="CreateOneScenarioTestEnvFile" StopOnFirstFailure="true" />
+  </Target>
+
   <!--
-    We need to include all dlls in the runtime path as inputs to make it really incremental if we use the root folder,
+    We need to include all dlls in the runtime path as inputs to make it really incremental. If we use the root folder,
     if a dll is updated, the folder's timestamp is not updated, therefore skipped.
   -->
   <ItemGroup>
     <_RuntimeInputs Include="$(TestHostRootPath)**/*.dll" />
-  </ItemGroup>
 
-  <ItemGroup>
-    <HelixProperties Condition="'$(RuntimeFlavor)' != ''" Include="runtimeFlavor" Value="$(RuntimeFlavor)" />
+    <!-- Add the scenario TestEnv batch files -->
```

Comment:
Did we needed to create an extra script file? I think we could have just used HelixPreRunCommands to set the environment.